### PR TITLE
Add support for GeoPoint and GeoLineString

### PR DESCRIPTION
**Pull request by guilhermeleobas**
Requires https://github.com/heavyai/heavydb-internal/pull/7126

This pull request adds support for three HeavyDB Geo column types (`GeoLineString`, `GeoPoint` and `Point2D`) and common operations:

- `is_null`
- `set_null`
- `get_item` / `operator.__getitem__`
- `set_item` / `operator.__setitem__`
- `get_n_of_values`
- `__len__`

### Structure of each type:

```c
struct ColumnGeo[LineString/Point] {
  int8_t* flatbuffer;
  int64_t size;
};

struct GeoLineString {
  int8_t* flatbuffer_;
  int64_t index1;
  int64_t index2;
  int64_t index3;
  int64_t index4;
  int32_t n_;
}

struct Point2D {
  float64 x, y;
};
```

### Caveats

**Parser doesn’t handle template specialization**

Current parser implementation doesn’t handle subtype alias. So, `Column<GeoPoint>` isn’t redirected to `HeavyDBColumnGeoPointType`, but rather `HeavyDBColumnType`. To circumvent this limitation, there’s a short-circuit in `HeavyDBColumnType::postprocess_type` which redirects the implementation to the correct type.

### CODE REVIEW

File: rbc/heavydb/table_function_manager.py [hunk 1]
```diff
@@ -154,6 +167,31 @@ def codegen(context, builder, sig, args):
     return sig, codegen
 
 
+@extending.intrinsic
+def mgr_set_output_item_(typingctx, mgr, col_idx, value):
+    sig = types.void(mgr, col_idx, value)
+
+    target_info = TargetInfo()
+    # XXX: Check if heavydb 6.3 has this function
```
Comment (pearu) [inline rbc/heavydb/table_function_manager.py]: 6.3 is not going to be released, so no.

**Review by pearu — COMMENTED** [review]
I did my first reading on this PR, nice work, @guilhermeleobas!

Before approving or requesting changes, I'll try it out and look into the typesystem issue.

File: rbc/heavydb/column_geolinestring.py [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+    sig = nb_types.int64(x)
+
+    def codegen(context, builder, sig, args):
+        [col] = args
+        return builder.extract_value(col, [1])
+
+    return sig, codegen
+
+
+@extending.overload_method(ColumnGeoLineStringPointer, 'getNofValues')
```
[the remark this answers is not in the document]
Comment (guilhermeleobas) [reply rbc/heavydb/column_geolinestring.py:171]: Changed!

File: rbc/heavydb/table_function_manager.py (hunk 1 above)
Comment (guilhermeleobas) [reply rbc/heavydb/table_function_manager.py]: We don't have 6.3 OSS, but do have EE, no?

**Comment by guilhermeleobas** [comment]
> Before approving or requesting changes, I'll try it out and look into the typesystem issue.

Ok. I'll edit the first comment to include a description of what I did and what is currently supported. And about the typesystem issue, we don't have to address it right now, as the workaround I did works.

File: rbc/heavydb/table_function_manager.py (hunk 1 above)
Comment (pearu) [reply rbc/heavydb/table_function_manager.py]: No, according to
https://releases.heavy.ai/ee/apt/pool/cpu/h/heavyai/

File: rbc/heavydb/remoteheavydb.py [hunk 3]
```diff
@@ -744,6 +749,7 @@ def load_table_columnar(self, table_name, **columnar_data):
                         arr_nulls = [v is None for v in arr]
                         if True in arr_nulls:
                             # TODO: null support for text
+                            # TODO: null support for Geo Types
```
Comment (guilhermeleobas) [inline rbc/heavydb/remoteheavydb.py:752]: This only applies to arrays<Geo*> types. Do we support Array with geo types?

Comment (pearu) [reply rbc/heavydb/remoteheavydb.py:752]: Not yet.

**Comment by guilhermeleobas** [comment]
Closing it in favor of #537

**Pull request closed without merging**